cluster-autoscaler/Dockerfile

```diff
@@ -11,9 +11,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG BASEIMAGE=gcr.io/distroless/static:nonroot-amd64
-FROM $BASEIMAGE
+FROM --platform=$BUILDPLATFORM golang:1.23 as builder
+
+WORKDIR /workspace
+
+COPY . .
+
+ARG GOARCH
+ARG LDFLAGS_FLAG
+ARG TAGS_FLAG
+
+RUN CGO_ENABLED=0 GOOS=linux go build -o cluster-autoscaler-$GOARCH $LDFLAGS_FLAG $TAGS_FLAG
+FROM gcr.io/distroless/static:nonroot
+ARG GOARCH
+COPY --from-builder cluster-autoscaler-$GOARCH /cluster-autoscaler
```

Comment:
```suggestion
COPY --from=builder cluster-autoscaler-$GOARCH /cluster-autoscaler
```

Reply:
should the target filepath also be prepended w/ the `/workspace` directory?

Reply:
no, just the source, you want the built binary in the root directory of the output container

Reply:
Right, so the `/workspace` directory is a prow locale that holds the checked out git source

I suppose the `WORKDIR /workspace` pragma isn't inherited by the build stage after `go build`

Reply:
yeah exactly, that's my understanding too

Reply:
done, stand by for squash commits